packages/storage-driver-s3/src/index.ts

```diff
@@ -39,12 +39,16 @@ export class DriverS3 implements Driver {
 	private serverSideEncryption: string | undefined;
 
 	constructor(config: DriverS3Config) {
-		const s3ClientConfig: S3ClientConfig = {
-			credentials: {
+		const s3ClientConfig: S3ClientConfig = {};
+
+		if ((config.key && !config.secret) || (config.secret && !config.key)) {
```

Comment:
Thoughts on simplifying with `XOR` operator?

```suggestion
		if (config.key ^ config.secret) {
```

Reply:
I think bitwise operators are harder to read but defer to other reviewers to tell their opinion 😅

Reply:
I do like this approach but i have my doubts on the maintainability as the bitwise XOR operator is not very commonly used and may be tricky for new members jumping into the codebase.

Reply:
> Thoughts on simplifying with `XOR` operator?

Looks cleaner IMO 😄👍 Personally, I'd go with this while adding a comment to explain it.

Reply:
> Looks cleaner IMO 😄👍 Personally, I'd go with this while adding a short comment to explain it.

Hard disagree! If it needs a comment explaining what it's supposed to do, you're probably better off making it slightly more verbose and maintainable in the process 🙂

Reply:
> Hard disagree! If it needs a comment explaining what it's supposed to do, you're probably better off making it slightly more verbose and maintainable in the process 🙂

Honestly, the more verbose option here isn't much more readable, so I'd have thought to add a comment in both cases anyway... But after looking at the code again, I realize that it's actually pretty obvious what the `if` statement does because of the verbose `error` statement one line below.

> [...] If it needs a comment explaining what it's supposed to do, you're probably better off making it slightly more verbose and maintainable in the process 🙂

I totally agree with you in principle (big fan of clean code), but I do think it doesn't harm do add a comment from time to time. Especially because the code only tells you what it does and not why it is done that way.

Reply:
> I totally agree with you in principle (big fan of clean code), but I do think it doesn't harm to add a comment from time to time. Especially because the code only tells you what it does and not why it is done that way.

I'm not against comments in general! I do have a problem when a comment is needed to explain _what_ a line of JS is supposed to do though. If that's the case, I'd rather rework the JS itself to make it more human readable 🙂 

Comments explaining _why_ something is required or needed are very valuable (as long as they're maintained)

Reply:
I am very happy to hear that! 😃 You summarized it very well and I can say that I completely agree with you!